### PR TITLE
[Monitor] BREAKING CHANGE: `az monitor log-analytics`: Support new api-version `2022-10-01`

### PR DESCRIPTION
**Pull request by AllyW**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Monitor] `az monitor log-analytics workspace create`: Add `--identity-type` and `--user-assigned` arguments
[Monitor] `az monitor log-analytics workspace update`: Add `--identity-type` and `--user-assigned` arguments
[Monitor] `az monitor log-analytics workspace table`: Enable `--retention-time` to be workspace retention when setting as `-1`
[Monitor] `az monitor log-analytics workspace table`: Enable `--total-retention-time` to be workspace retention when setting as `-1`
[Monitor] `az monitor log-analytics workspace table search-job`: Add new command `cancel`
[Monitor] BREAKING CHANGE: `az monitor log-analytics workspace table search-job create`: Remove `schema` wrapper for `searchResults` in api response

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

Closed https://github.com/Azure/azure-cli/issues/23546
Closed https://github.com/Azure/azure-cli/issues/22909
Closed https://github.com/Azure/azure-cli/issues/21936

### CODE REVIEW

**Comment by yonzhan** [comment]
Monitor

File: src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/_create.py [hunk 1]
```diff
@@ -81,12 +106,14 @@ def _build_arguments_schema(cls, *args, **kwargs):
             options=["--ingestion-access"],
             arg_group="Properties",
             help="The public network access type to access workspace ingestion.",
+            default="Enabled",
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/_create.py:109]: Is it a breaking change from None to Enabled

File: src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/_create.py [hunk 2]
```diff
@@ -81,12 +106,14 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
             enum={"Disabled": "Disabled", "Enabled": "Enabled"},
         )
         _args_schema.query_access = AAZStrArg(
             options=["--query-access"],
             arg_group="Properties",
             help="The public network access type to access workspace query.",
+            default="Enabled",
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/_create.py:116]: Same concern here

File: src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_log_analytics_workspace.py [hunk 3]
```diff
@@ -528,9 +528,9 @@ def test_monitor_log_analytics_workspace_table(self, resource_group):
 
         self.cmd('monitor log-analytics workspace table search-job create -n {table2_name} -g {rg} --workspace-name {ws_name} --retention-time 50 --total-retention-time 80 --start-search-time "2021-08-01 05:29:18" --end-search-time "2021-08-02 05:29:18" --search-query "Heartbeat" --limit 1', checks=[
             self.check('name', '{table2_name}'),
-            self.check('schema.searchResults.query', 'Heartbeat'),
-            self.check('schema.searchResults.limit', 1),
-            self.check('schema.searchResults.sourceTable', "Heartbeat"),
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_log_analytics_workspace.py:533]: It seems the output schema is changed

File: src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/_create.py (hunk 1 above)
Comment (AllyW) [reply src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/_create.py:109]: it's enabled from old swagger definition, behavior are the same as before

File: src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/_create.py (hunk 2 above)
Comment (AllyW) [reply src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/_create.py:116]: it's enabled from old swagger definition, behavior are the same as before

File: src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_log_analytics_workspace.py (hunk 3 above)
Comment (AllyW) [reply src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_log_analytics_workspace.py:533]: yes, schema is removed here. just properties.searchResultes returned

**Review by kairu-ms — DISMISSED** [review]
LGTM. Recommend release this PR in next sprint which is a breaking change window.

File: src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_log_analytics_workspace.py (hunk 3 above)
Comment (kairu-ms) [reply src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_log_analytics_workspace.py:533]: Please add this output change in History note. It's a breaking change.

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py [hunk 4]
```diff
@@ -330,23 +332,27 @@ def load_arguments(self, _):
     with self.argument_context('monitor log-analytics workspace table') as c:
         c.argument('table_name', name_arg_type, help='Name of the table.')
         c.argument('workspace_name', options_list='--workspace-name')
-        c.argument('retention_in_days', type=int, options_list='--retention-time', help='The data table data retention in days, between 4 and 730. Setting this property to null will default to the workspace')
-        c.argument('total_retention_in_days', type=int, options_list='--total-retention-time', help='The table data total retention in days, between 4 and 2555. Setting this property to null will default to table retention.')
+        c.argument('retention_in_days', type=int, options_list='--retention-time', help='The table retention in days, between 4 and 730. Setting this property to -1 will default to the workspace retention.')
+        c.argument('total_retention_in_days', type=int, options_list='--total-retention-time', help='The table total retention in days, between 4 and 2556. Setting this property to -1 will default to table retention.')
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/_params.py:336]: This change should also be added in History Notes

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_nw_flow_log.py [hunk 5]
```diff
@@ -9,8 +9,8 @@
 
 class NWFlowLogScenarioTest(ScenarioTest):
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_nw_flow_log.py:10]:
```suggestion
class NWFlowLogScenarioTest(LiveScenarioTest):
```
Could we simply bypass the test cases like above?

**Review by evelyn-ys — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Review by calvinhzy — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**